ProblemGenerator/module_computerphile.py

```diff
@@ -0,0 +1,44 @@
+import random
+
+import Board.module_board
+from ProblemGenerator.module_problem_base_class import ProblemFinder
+
+
+class Computerphile(ProblemFinder):
+    """
+    This Algorithm is inspired by the youtube video from the Computerphile channel.
+    https://www.youtube.com/watch?v=G_UYXzGuqvM
+    """
+
+    def __init__(self, board: Board, seed: int, log_name: str):
+        super().__init__(board, seed, log_name)
+        self.board = board
+
+    def return_problem_solution(self):
+        for y in range(self.board.get_board_length()):
+            for x in range(self.board.get_board_length()):
+                if self.board.get_value(x, y) == -1 or self.board.get_value(x, y) is None:
+                    random_value_list = list(range(1, self.board.get_board_length() + 1))
+                    while len(random_value_list) != 0:
+                        n = random.choice(random_value_list)
+                        random_value_list.remove(n)
+                        if self.possible(y, x, n):
+                            self.board.set_value(x, y, n)
+                            if self.return_problem_solution():
```

Comment:
was macht diese if abfrage?

Reply:
ist das das zurückgehen wenn es später nicht klappt?

Reply:
angenommen das letzte element vor dem rekursiven aufruf war das letzte in der liste. wird also in diesem schleifen durchlauf gelöscht und die liste ist jetzt leer. dann kommt später der rekursive aufruf zurück. dann setze ich das auf -1 kann aber die schleife nicht nochmal betreten und gebe True zurück obwohl das feld leer ist.

Reply:
Das wäre richtig, weil es dann keine Lösung für das Sudoku gibt und er an diesem Feld nichts einsetzen kann. 
Aber ich glaube der Algorithmus würde bei einem unlösbaren Problem ewig laufen, dass muss ich mir morgen nochmal genauer anschauen.

Reply:
was ich meinte ist, dass der rekursive aufruf dann weiter macht aber in dem feld ne -1 drinn steht oder nicht? also könnte er dann theoretisch ne lösung finden wo leere felder drin sind aber er gibt trotzdem zurück das er ne lösung hat.

Reply:
Ich hab jetzt eine Änderung gepushed, welche True oder False zurück gibt, ob eine Lösung gefunden wurde.
Ich werde noch ein Ticket erstellen dass, das in den anderen Algorithmen übernommen wird.

Reply:
Neues Issues #16

Reply:
Also resolve und merge ich das hier jetzt?